pkg/status/controller.go

```diff
@@ -328,8 +328,16 @@ func updateObjectStatus(ctx context.Context, objRef *util.SourceRef, status map[
 
 	if objRef.Namespace == "" {
 		_, err = wdsDynClient.Resource(gvr).UpdateStatus(ctx, unstrObj, metav1.UpdateOptions{})
+		// if resource not found it may mean no status subresource - try to update the whole resource
+		if errors.IsNotFound(err) {
+			_, err = wdsDynClient.Resource(gvr).Update(ctx, unstrObj, metav1.UpdateOptions{})
+		}
 	} else {
 		_, err = wdsDynClient.Resource(gvr).Namespace(objRef.Namespace).UpdateStatus(ctx, unstrObj, metav1.UpdateOptions{})
+		// if resource not found it may mean no status subresource - try to update the whole resource
+		if errors.IsNotFound(err) {
+			_, err = wdsDynClient.Resource(gvr).Namespace(objRef.Namespace).Update(ctx, unstrObj, metav1.UpdateOptions{})
```

Comment:
In addition to previous comment, it may be more efficient just to update the status section with a patch rather than an update of the whole object. For example, I was experimenting with:

```
patchBytes, _ := json.Marshal(statusPatch.Object)
			_, err = wdsDynClient.Resource(gvr).Namespace(objRef.Namespace).Patch(ctx, statusPatch.GetName(), types.MergePatchType, patchBytes, metav1.PatchOptions{})
```

where:

```
func createStatusPatch(unstrObj *unstructured.Unstructured, status map[string]interface{}) *unstructured.Unstructured {
	patchedObj := &unstructured.Unstructured{}
	patchedObj.SetAPIVersion(unstrObj.GetAPIVersion())
	patchedObj.SetKind(unstrObj.GetKind())
	patchedObj.SetName(unstrObj.GetName())
	patchedObj.SetNamespace(unstrObj.GetNamespace())
	patchedObj.Object["status"] = status
	return patchedObj
}
```

Reply:
Is `patch` API call more efficient than `update` API call?

Reply:
The difference here is that using the update method the whole object (spec and status) is retrieved from the lister, status is updated on that object and then the update updates the whole object (spec and status). The approach I proposed above only sets API version, kind, name and namespace and the status to make the patch. So I expect it to be more efficient as less content is sent over the wire, and it seems also cleaner to me. On the other hand, this was not an issue for the status subresource as in that case only status is being updated.

Reply:
Switched to use `patch` instead of `update`.